src/apps/logging/logging.cpp

```diff
@@ -257,6 +259,81 @@ namespace loggingapp
       };
       // SNIPPET_END: log_record_text
 
+      auto& historical_state = context.get_historical_state();
+      auto get_historical = [this, &historical_state](ccf::RequestArgs& args) {
+        const auto params =
+          nlohmann::json::parse(args.rpc_ctx->get_request_body());
+        const auto in = params.get<LoggingGetHistorical::In>();
+
+        // Check that the requested transaction ID is committed
+        {
```

Comment:
This seems like boilier plate that will be written for every handler that implements a historical query. Can this be moved into a helper method?

Reply:
In part: If we pass the target view and seqno in headers rather than the body they can be parsed by the framework/helper method. But I'd like to get the structure in in this PR, and work on how cleanly we expose it to apps later.